src/preamble.js

```diff
@@ -780,10 +780,23 @@ var memoryInitializer = null;
 #include "memoryprofiler.js"
 #endif
 
-#if PTHREAD_POOL_SIZE > 0 && PTHREAD_POOL_DELAY_LOAD != 1
-// To work around https://bugzilla.mozilla.org/show_bug.cgi?id=1049079, warm up a worker pool before starting up the application.
-if (!ENVIRONMENT_IS_PTHREAD) addOnPreRun(function() { if (typeof SharedArrayBuffer !== 'undefined') { addRunDependency('pthreads'); PThread.allocateUnusedWorkers({{{PTHREAD_POOL_SIZE}}}, function() { removeRunDependency('pthreads'); }); }});
+#if USE_PTHREADS && PTHREAD_POOL_SIZE
+
+var pthreadPoolSize = Module['pthreadPoolSize'] || {{{ PTHREAD_POOL_SIZE }}};
```

Comment:
If one wanted to set `Module['pthreadPoolSize'] = 0;` to disable pthread pool, this has the effect of using `{{{PTHREAD_POOL_SIZE}}}` instead?

Reply:
Oh, good point. Perhaps this should be more careful then.

Another thought I had meanwhile, if we're considering delaying this PR anyhow as discussed above - perhaps `PTHREAD_POOL_SIZE` could be a *string* - then the user could just set it to `navigator.hardwareConcurrency` directly, what do you think?

Reply:
I like that idea - they could then set it to `Module['pthreadPoolSize']` as well to customize.